### PR TITLE
Restore from database

### LINKED ISSUE #264: Restore from database
When starting the backend, all relevant data should be loaded from the database.

Depends on #263.

### PR DESCRIPTION
**Pull request by ClFeSc**
Closes #264 

Depends on #350 

TODOs:
- [x] A test fails

### CODE REVIEW

File: shared/src/store/reducer-error.ts [hunk 1]
```diff
@@ -1,10 +1,16 @@
 /**
  * This error gets thrown if an error in a reducer function occurs because the action isn't compatible to the state.
  */
+// TODO: This comment is now outdated.
```
Comment (Dassderdie) [inline shared/src/store/reducer-error.ts:4]: Just remove the `and we don't need a stack trace,` below

**Review by Dassderdie — COMMENTED** [review]
Not tested out yet. Only small stuff

File: backend/src/utils/user-readable-id-generator.ts [hunk 2]
```diff
@@ -44,4 +44,12 @@ export class UserReadableIdGenerator {
     public static freeAll() {
         this.generatedIds.clear();
     }
+
+    /**
+     * Notes all provided {@link ids} as used.
+     * @param ids The ids to lock
```
Comment (anonym-HPI) [inline backend/src/utils/user-readable-id-generator.ts:50]:
```suggestion
     * @param ids to lock
```

**Review by anonym-HPI — CHANGES_REQUESTED** [review]
Tested it (killing dfm container) and starting it again, seems to be working.

**Review by Dassderdie — APPROVED** [review]

**Review by anonym-HPI — APPROVED** [review]

**Pull request merged**